File: LC169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0, ele;

        for(auto i : nums){
            if(!count)
                ele = i;
            count += (i == ele)? 1 : -1;
        }
        
        return ele;
    }

};

File: LC169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ele = None
        count = 0
        
        for i in nums:
            if not count:   # if count == 0
                ele = i
            
            count += 1 if i == ele else -1

        return ele
